surfsense_backend/app/tasks/podcast_tasks.py

```
@@ -24,73 +26,177 @@ async def generate_chat_podcast(
+            {"error_type": "SQLAlchemyError"}
+        )
+        raise db_error
+    except Exception as e:
+        await session.rollback()
+        await task_logger.log_task_failure(
+            log_entry,
+            f"Unexpected error during podcast generation for chat {chat_id}",
+            str(e),
+            {"error_type": type(e).__name__}
+        )
+        raise RuntimeError(f"Failed to generate podcast for chat {chat_id}: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)